src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const copy = { ...state };
+
+  actions.forEach(action => {
+    switch (action.type) {
+      case ('addProperties'):
+        Object.assign(copy, action.extraData);
+        break;
+
+      case ('removeProperties'):
+        action.keysToRemove.forEach(rem => {
+          delete copy[rem];
+        });
+        break;
+
+      case ('clear'):
+        for (const key in copy) {
+          delete copy[key];
+        }
+        break;
+
+      default: return 'Please, enter the valid data';
```

Comment:
```suggestion
      default: return 'Unknown action type';
```

and in this case better to use [throw new Error](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw)